dbms/src/Flash/Coprocessor/DAGQueryBlockInterpreter.cpp

```diff
@@ -695,8 +695,11 @@ void DAGQueryBlockInterpreter::handleJoin(const tipb::Join & join, DAGPipeline &
 
     // add a HashJoinBuildBlockInputStream to build a shared hash table
     size_t stream_index = 0;
+    auto get_stream_index = [&stream_index, &join_build_concurrency]() {
+        return (stream_index++) % join_build_concurrency;
+    };
     right_pipeline.transform(
-        [&](auto & stream) { stream = std::make_shared<HashJoinBuildBlockInputStream>(stream, join_ptr, stream_index++, log->identifier()); });
+        [&](auto & stream) { stream = std::make_shared<HashJoinBuildBlockInputStream>(stream, join_ptr, get_stream_index(), log->identifier()); });
```

Comment:
The name `stream_index` is kind of indicating it should be unique between different `HashJoinBuildBlockInputStream`, I prefer to change the name or move the `stream_index%join_build_concurrency` to `Join.cpp`

Reply:
But `join.insertFromBlock` will do `stream_index%join_build_concurrency` every time.
How about a different name? Like `join_build_pool_id`?

Reply:
LGTM